lib/m2ts/m2ts.js

```diff
@@ -343,84 +390,101 @@ ElementaryStream = function() {
-          streamType = 'video';
-          break;
-        case StreamTypes.ADTS_STREAM_TYPE:
-          stream = audio;
-          streamType = 'audio';
-          break;
-        case StreamTypes.METADATA_STREAM_TYPE:
-          stream = timedMetadata;
-          streamType = 'timed-metadata';
-          break;
-        default:
-          // ignore unknown stream types
-          return;
-        }
+    if(typeof data !== 'object' ||
+       !data.type) {
+      this.error('Data with no type passed to ElementaryStream');
+      return;
+    }
+    if(data.type === 'pat') {
+      // we have to wait for the PMT to arrive as well before we
+      // have any meaningful metadata
+      return;
+    }
+    if(data.type === 'pes') {
+      var stream, streamType;
+
+      switch (data.streamType) {
+      case StreamTypes.H264_STREAM_TYPE:
+        stream = video;
+        streamType = 'video';
+        break;
+      case StreamTypes.ADTS_STREAM_TYPE:
+        stream = audio;
+        streamType = 'audio';
+        break;
+      case StreamTypes.METADATA_STREAM_TYPE:
+        stream = timedMetadata;
+        streamType = 'timed-metadata';
+        break;
+      case undefined:
+        this.error('Found an undefined pes stream type');
+        return;
+      default:
+        this.error('Found an unknown pes stream type ' + data.streamType.toString(16));
+        return;
+      }
 
-        // if a new packet is starting, we can flush the completed
-        // packet
-        if (data.payloadUnitStartIndicator) {
-          flushStream(stream, streamType);
-        }
+      // if a new packet is starting, we can flush the completed
+      // packet
+      if (data.payloadUnitStartIndicator) {
+        flushStream(stream, streamType);
+      }
 
-        // buffer this fragment until we are sure we've received the
-        // complete payload
-        stream.data.push(data);
-        stream.size += data.data.byteLength;
-      },
-      pmt: function() {
-        var
-          event = {
-            type: 'metadata',
-            tracks: []
-          },
-          programMapTable = data.programMapTable,
-          k,
-          track;
-
-        // translate streams to tracks
-        for (k in programMapTable) {
-          if (programMapTable.hasOwnProperty(k)) {
-            track = {
-              timelineStartInfo: {
-                baseMediaDecodeTime: 0
-              }
-            };
-            track.id = +k;
-            if (programMapTable[k] === m2tsStreamTypes.H264_STREAM_TYPE) {
-              track.codec = 'avc';
-              track.type = 'video';
-            } else if (programMapTable[k] === m2tsStreamTypes.ADTS_STREAM_TYPE) {
-              track.codec = 'adts';
-              track.type = 'audio';
+      // buffer this fragment until we are sure we've received the
+      // complete payload
+      stream.data.push(data);
+      stream.size += data.data.byteLength;
+    } else if (data.type === 'pmt') {
```

Comment:
You have an `else` here for this condition. Though the first two conditions short-circuit with `return`, it is the unofficial style that all the conditions in a block like this should be `else if` for consistency and safety. That way if someone changes one of the earlier conditional blocks to no longer short-circuit, it doesn't break things.